scipy/stats/tests/test_device_dtype.py

```diff
@@ -68,14 +69,21 @@ def test_directional_stats(dtype, xp, devices):
         assert res.mean_resultant_length.dtype == dtype
 
 
-@pytest.mark.parametrize('method', ['vasicek', 'van es', 'ebrahimi', 'correa', 'auto'])
+@pytest.mark.parametrize('method', [
+    "vasicek",
+    "van es",
+    pytest.param(
+        "correa",
+        marks=[
+            skip_xp_backends("array_api_strict", reason="Invalid fancy indexing"),
+            skip_xp_backends("dask.array", reason="Invalid fancy indexing"),
+        ],
+    ),
+    "ebrahimi",
+    "auto",
+])
```

Comment:
@rgommers In https://github.com/scipy/scipy/pull/22856#discussion_r2117586032 it sounded like you were OK with either syntax for skipping tests. Here, `parametrize` is less concise. Is this change necessary?